src/main/java/com/sleekbyte/tailor/Tailor.java

```diff
@@ -123,83 +184,142 @@ public static CommonTokenStream getTokenStream(File inputFile) throws IOExceptio
-                listeners.add(new MaxLengthListener(printer, constructLengths, enabledRules));
-                listeners.add(new MinLengthListener(printer, constructLengths, enabledRules));
-                DeclarationListener decListener = new DeclarationListener(listeners);
-                listeners.add(decListener);
-
-                ParseTreeWalker walker = new ParseTreeWalker();
-                for (SwiftBaseListener listener : listeners) {
-                    // The following listeners are used by DeclarationListener to walk the tree
-                    if (listener instanceof ConstantNamingListener || listener instanceof KPrefixListener) {
-                        continue;
-                    }
-                    walker.walk(listener, tree);
+        files.parallelStream().forEach(
+            file -> {
+                try {
+                    Optional<CommonTokenStream> tokenStream = getTokenStream(file);
+                    Optional<TopLevelContext> tree = getParseTree(tokenStream);
+                    analyzeFile(file, tokenStream, tree, formatter, maxSeverity, constructLengths, enabledRules);
+                    printIfFormatAllows(".");
+                } catch (ErrorListener.ParseException e) {
+                    printIfFormatAllows("S");
+                    Printer printer = new Printer(file, maxSeverity, formatter);
+                    printer.setShouldPrintParseErrorMessage(true);
+                    printersForAllFiles.add(printer);
+                    numSkippedFiles.incrementAndGet();
+                } catch (CLIArgumentParserException e) {
+                    handleCLIException(e, configuration);
                 }
-                try (FileListener fileListener = new FileListener(printer, inputFile, constructLengths, enabledRules)) {
-                    fileListener.verify();
+            });
+        printIfFormatAllows(String.format("%n"));
+
+        printersForAllFiles.forEach(printer -> {
+                try {
+                    printer.printAllMessages();
+                } catch (IOException e) {
+                    handleIOException(e);
                 }
+            });
 
-                numErrors += printer.getNumErrorMessages();
-                numWarnings += printer.getNumWarningMessages();
-            }
-        }
+        formatter.displaySummary(fileNames.size(), numSkippedFiles.get(), numErrors.get(), numWarnings.get());
+        handleErrorViolations(formatter, numErrors.get());
+    }
 
-        if (formatter != null) {
-            formatter.displaySummary(fileNames.size(), numSkippedFiles, numErrors, numWarnings);
-            // Non-zero exit status when any violation messages have Severity.ERROR, controlled by --max-severity
-            ExitCode exitCode = formatter.getExitStatus(numErrors);
-            if (exitCode != ExitCode.SUCCESS) {
-                System.exit(exitCode.ordinal());
+    private void printIfFormatAllows(String str) {
```

Comment:
thoughts on just adding a print function to the `Formatter` abstract class and calling it? 

different formatters could implement it differently, I think its a cleaner solution (also removes some more code from this file :stuck_out_tongue:)